File: frontend/src/api/stats.ts
import request from './request'

export interface DailyCount {
  date: string
  count: number
}

export interface StatsOverview {
  total_scans: number
  allergen_alerts: number
  today_scans: number
  avg_score: number
  recent_days: DailyCount[]
}

export const getStatsApi = () => request.get<StatsOverview>('/stats/overview').then((r) => r.data)

File: backend/app/schemas/stats.py
"""首页仪表板统计模型"""
from pydantic import BaseModel


class DailyCount(BaseModel):
    date: str
    count: int


class StatsOverview(BaseModel):
    total_scans: int = 0
    allergen_alerts: int = 0
    today_scans: int = 0
    avg_score: float = 0
    recent_days: list[DailyCount] = []
